src/redlock.rs

```diff
@@ -141,59 +177,45 @@ impl RedLock {
         }
     }
 
-    /// Unlock the given lock.
-    ///
-    /// Unlock is best effort. It will simply try to contact all instances
-    /// and remove the key.
-    pub async fn unlock(&self, lock: &Lock<'_>) {
-        join_all(
-            self.servers
-                .iter()
-                .map(|client| self.unlock_instance(client, &lock.resource, &lock.val)),
-        )
-        .await;
-    }
-
-    /// Acquire the lock for the given resource and the requested TTL.
-    ///
-    /// If it succeeds, a `Lock` instance is returned,
-    /// including the value and the validity time
-    ///
-    /// If it fails. `None` is returned.
-    /// A user should retry after a short wait time.
-    pub async fn lock(&self, resource: &[u8], ttl: usize) -> Result<Lock<'_>, RedLockError> {
-        let val = self.get_unique_lock_id().unwrap();
-
+    // Can be used for creating or extending a lock
+    async fn lock_with_duration<'a: 'b, 'b, T, Fut>(
+        &'a self,
+        resource: &[u8],
+        value: &[u8],
+        duration: usize,
+        lock: T,
+    ) -> Result<Lock<'_>, RedLockError>
```

Comment:
I would rename this method `exec_or_retry` and the lock field `exec` since it's purpose is to exec a closure that takes a client as parameter with a retry mechanism.

What is your opinion?